File: go-ut-bench/datasets/go/go_code_files_self_contained/simple_function/simple_function_000.go
package main

func CalculateFibonacci(n int) int {
	if n < 0 {
		panic("n must be non-negative")
	}
	if n == 0 {
		return 0
	}
	if n == 1 {
		return 1
	}
	prev, curr := 0, 1
	for i := 2; i <= n; i++ {
		next := prev + curr
		prev = curr
		curr = next
	}
	return curr
}

File: go-ut-bench/datasets/cpp/cpp_code_files_self_contained/simple_function/simple_function_000.cpp
#include <stdexcept>

int calculateFibonacci(int n) {
    if (n < 0) {
        throw std::invalid_argument("n must be non-negative");
    }
    if (n == 0) {
        return 0;
    }
    if (n == 1) {
        return 1;
    }
    int prev = 0;
    int curr = 1;
    for (int i = 2; i <= n; i++) {
        int next = prev + curr;
        prev = curr;
        curr = next;
    }
    return curr;
}